pkg/controller/datavolume-controller.go

```diff
@@ -352,20 +364,35 @@ func (c *DataVolumeController) syncHandler(key string) error {
 		return errors.Errorf(msg)
 	}
 
+	// expectations prevent us from creating multiple pods. An expectation forces
+	// us to observe a pod's creation in the cache.
 	needsSync := c.pvcExpectations.SatisfiedExpectations(key)
+
 	if !exists && needsSync {
-		newPvc, err := newPersistentVolumeClaim(dataVolume)
-		if err != nil {
-			return err
-		}
-		c.pvcExpectations.ExpectCreations(key, 1)
-		pvc, err = c.kubeclientset.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Create(newPvc)
-		if err != nil {
-			c.pvcExpectations.CreationObserved(key)
-			return err
-		}
-		if canUpdateProgress(newPvc.Annotations) {
-			go c.scheduleProgressUpdate(dataVolume.Name, dataVolume.Namespace, pvc.GetUID())
+		snapshotClassName := c.getSnapshotClassForSmartClone(dataVolume)
+		if snapshotClassName != "" {
+			klog.V(3).Infof("Smart-Clone via Snapshot is available with Volume Snapshot Class: %s", snapshotClassName)
+			newSnapshot := newSnapshot(dataVolume, snapshotClassName)
+			c.pvcExpectations.ExpectCreations(key, 1)
```

Comment:
So I don't think you need the expectations at all, let me explain why. Expectations are like a lock counter, they are needed to account for the request for creation of pods and them actually being created. But they are only needed if the pods names are not unique, like in the case of an importer or cloner pod. However the snapshot name is derived from the data volume name, which by itself is guaranteed to be unique, and thus the snapshot is guaranteed to be unique, so we don't need the expectations to count if the snapshots are being created. I had the same thing when I worked on the scratch space and @mhenriks rightfully pointed out I didn't need them. I think the same is true for snapshots.

Reply:
ok, got it. removed.